src/main/java/org/jenkinsci/plugins/workflow/support/steps/WorkspaceStepExecution.java

```diff
@@ -35,6 +35,9 @@ public boolean start() throws Exception {
         if (!(job instanceof TopLevelItem)) {
             throw new Exception(job + " must be a top-level job");
         }
+        if (!getContext().hasBody()) {
+            throw new Exception("ws must be called with a body");
```

Comment:
I'd use `AbortException` since we don't care about the stack trace, and tweak the message slightly:

```suggestion
            throw new AbortException("The ws step must be called with a body"); // Requires an import of hudson.model.AbortException
```